### ISSUE TITLE
coupe of things --> possible already?

### ISSUE THREAD

**Issue by Lykdraft** [kmatheussen/radium#1209]
couple of little things here where I'd like to know if that's possible already:

- close VST/VSTi Window without clicking. (i.e.: escape key or a toggle command)

- load/save block, load/save track ---> hotkeys possible?

- select Seqtrack (itself) via keycommands / select Blocks **on** Seqtrack via keycommands.
(next/prev block on seqtrack. The normal playlist next/prev won't work. When I start
playback the track edit view snaps back to view the selected/highlighted block in the seqtrack, a bit confusing.)

- possible to change the red rectangles around all blocks on the selected seqtrack?
(I've looked through the color config, but couldn't find the entry to change that. It's very much "in your face" so to speak. )

Thanks.
tL.

**Comment by kmatheussen** [comment]
On Fri, May 10, 2019 at 3:42 AM Lykdraft <notifications@github.com> wrote:
>
> couple of little things here where I'd like to know if that's possible already:
>
> close VST/VSTi Window without clicking. (i.e.: escape key or a toggle command)
>

This should work:

ra.evalScheme "(ra:hide-instrument-gui (ra:get-current-instrument))"


> load/save block, load/save track ---> hotkeys possible?
>

Yeah, the functions are

ra.loadTrack
ra.saveTrack
ra.loadBlock
ra.saveBlock


> select Seqtrack (itself) via keycommands / select Blocks on Seqtrack via keycommands.
> (next/prev block on seqtrack. The normal playlist next/prev won't work.

I'm adding this really soon. With separate keyboard forcus for the
sequencer, it's natural to add all these things.

> When I start
> playback the track edit view snaps back to view the selected/highlighted block in the seqtrack, a bit confusing.)
>
What's "track edit view"?

> possible to change the red rectangles around all blocks on the selected seqtrack?
> (I've looked through the color config, but couldn't find the entry to change that. It's very much "in your face" so to speak. )

Red rectangels around selected seqblocks? I don't understand what you mean...

**Comment by kmatheussen** [comment]
> Red rectangels around selected seqblocks? I don't understand what you mean...

You mean the transparent orange-like color for seqblocks currently being moved?
That's "Sequencer block multiselected background color".

**Comment by Lykdraft** [comment]
Red Rectangles around ALL blocks in the current seqtrack:
![red rectn](https://user-images.githubusercontent.com/18437561/57525991-8514d100-732c-11e9-8ccc-f0f17289f64d.gif)

**Comment by Lykdraft** [comment]
>  When I start
>  playback the track edit view snaps back to view the selected/highlighted block in the seqtrack, a bit confusing.)

> What's "track edit view"?

Track Edit View = upper left area, track edit window.

Heres a .gif about what I mean, however it might be not that clear.
![snapback](https://user-images.githubusercontent.com/18437561/57526364-85619c00-732d-11e9-9ca2-f15106ad9bfb.gif)

I also made a short q.mov, if it's not clear from the .gif what I am talking about.

https://www.dropbox.com/s/x95l797a25dycp9/snapback.mov?dl=0
(8mb)
You can see that when Radium is in **Stop**, a **different** block is shown in the track editor
than when Radium is running. When I select the correct block via mouse all is good. But this adds
a level of micromanagement to everything that is rather odd. Especially new users might get very confused by this?


_(I think this depends somehow on what is currently selected by mouse
in the seqtrack  and on the other hand what is currently selected normally in the playlist, but I am
not sure, couldn't really figure it out 100% yet.This is also why I asked for the Select SeqTrack by hotkey. To get around this without the need to go to the sequencer and click the correct track.)_

**Comment by kmatheussen** [comment]
On Fri, May 10, 2019 at 2:04 PM Lykdraft <notifications@github.com> wrote:

> Red Rectangles around ALL blocks in the current seqtrack:
>

Oh, that's a border to show that this seqblock holds the current block.
I've looked up the code, and it actually uses the editor cursor color.
I've been meaning to make this border thinner and brighter for a while now.
It doesn't look good.

**Comment by kmatheussen** [comment]
Maybe just show a circle or something next to the block name instead
of painting a border. It's confusing to have two different types of
current seqblock border.

**Comment by kmatheussen** [comment]
I didn't see your last points. (maybe you edited after I answered?)

> hotkey for opening mixer window? (especially mixer from selected soundobjects)

Yes:

```
F8: ra.evalScheme "(ra:show-mixer-strips2 2 (ra:get-selected-instruments))"
```
("2" is to select number of rows)


> hotkey for "F" (Sequence Track Window Full Screen View)

To a certain degree. You can use ra.hideUpperPartOfMainWindow and ra.showUpperPartOfMainWindow, but these two functions don't remember heights of the widgets. It might be good enough though.


> I want to remap the CTRL-J (Jump to line) to the "." on the numblock.
> However I don't know how NUMBLOCK-"." is defined.

I don't understand....

**Comment by Lykdraft** [comment]
> I want to remap the CTRL-J (Jump to line) to the "." on the numblock.
However I don't know how NUMBLOCK-"." is defined.

> I don't understand....

Right now "Jump To Line Number XYZ ", which gives you the menu, is mapped to L_CTRL-J. 
I would like to **remap** this L_CTRL-J hotkey to the "**dot**" between the **0** and the **ENTER** Key on my numpad (a.k.a. 10keys a.k.a numbersblock right on the side of my computerkeyboard)

but I don't know how the "dot" must be defined as a keybinding.

**Comment by kmatheussen** [comment]
Ah. I thought you wanted to extend CTRL-J to jump to a place called
NUMBLOCK-".".  :-)
The key is called KP_DOT:
http://users.notam02.no/~kjetism/radium/documentation/keybindings_framed.html

**Comment by kmatheussen** [comment]
It's called keypad, by the way. At least that seems to be the most common
name for it.

**Comment by Lykdraft** [comment]
Lol. "A Place Called Numblock..." Good songname.. :)


> http://users.notam02.no/~kjetism/radium/documentation/keybindings_framed.html
Whoa, where does this come from? Never saw that before. That's a good cheat sheet.

**Comment by Lykdraft** [comment]
Alright, so I think we are all good here, except the red-rectangle box thing around the blocks
in the sequencer? Shall I make a separate thread for that?

**Comment by kmatheussen** [comment]
Just keep this one open. I'll remember the remaining point from this thread.

**Comment by kmatheussen** [comment]
On Sat, May 11, 2019 at 8:50 PM Lykdraft <notifications@github.com> wrote:

> Lol. "A Place Called Numblock..." Good songname.. :)
>
>
> http://users.notam02.no/~kjetism/radium/documentation/keybindings_framed.html
> Whoa, where does this come from? Never saw that before. That's a good
> cheat sheet.
>
>
It's in the manual. (Help -> Manual)

**Comment by Teteros** [comment]
Another random thought related to borders:

![main editor window](https://user-images.githubusercontent.com/5067989/57575840-51c86400-744a-11e9-878c-427432621a80.png)

The border indicating currently selected track would look better imo if they don't extend past it.
Edited example:
![edited border](https://user-images.githubusercontent.com/5067989/57575991-13cd3f00-744e-11e9-9042-a0cd9aed853a.png)

**Comment by kmatheussen** [comment]
On Fri, May 10, 2019 at 8:28 AM Kjetil Matheussen
<k.s.matheussen@gmail.com> wrote:
>
> This should work:
>
> ra.evalScheme "(ra:hide-instrument-gui (ra:get-current-instrument))"
>

A show/hide entry is also in the "Windows" menu. It's defined like this:
ra.evalScheme "(show/hide-instrument-gui)"

**Comment by kmatheussen** [comment]
On Sun, May 12, 2019 at 1:37 AM Teteros <notifications@github.com> wrote:
>
> Another random thought related to borders:
>
> The border indicating currently selected track would look better imo if they don't extend past it.

I'll think about it. I'm not seeing why this is better though, and
it's also a bit harder to implement.

**Comment by Lykdraft** [comment]
I agree. It's looking better on "paper" but from a practical perspective, it is actually quite nice that the border extends to the track names above. This helps with a busy track editor.
As long as there's not much going on this would be fine. Later on, not so much. 

Because your eyes need to retrack from the (then moving) border to the trackname above. With the borders
extending to the tracknames above it is very easy to keep focus. So I would argue to leave it that way.

_(Also when done like in the picture shown, the top and bottom border would disapper when scrolling up/down. Most of the time you would only see the top border or bottom border or none of 'em. This is not good. And when the big track mute X's come into play, this also is more homogenous with the design.)_
